symbolic-base/src/ZkFold/Symbolic/Data/Ed25519.hs

```diff
@@ -49,14 +44,14 @@ instance
     -- It will need additional checks in pointDouble because of the denominator becoming zero, though.
     -- TODO: Think of a better solution
     --
-    pieces Inf         = hliftA2 V.append <$> pieces (zero :: UInt 256 'Auto c) <*> pieces (zero :: UInt 256 'Auto c)
+    pieces Inf         = hliftA2 V.append <$> pieces (zero :: FFA Ed25519_Base c) <*> pieces (zero :: FFA Ed25519_Base c)
```

Comment:
`retore (pieces Inf)` is not `Inf`

Reply:
There is no way to return `Inf` with the current interface. `bool` relies on `restore` which would rely on `bool` itself if we were to perform a check if the point is (0, 0). The only solution which will not use such things is completely removing Ed25519 from Symbolic and treating it separately each time we want to use it

Reply:
Ok, I guess we could leave it as is for now. In general, Symbolic cannot normally support sum types. I think at some point, we should change the `Point` type.

Reply:
I'd honestly define a separate "Symbolic Ed25519" datatype which is just a pair $(x, y)$ of affine coordinates
There's no need for it to be a `Point`, and we can reuse `addPoints` etc. from curves by just putting coordinates into `Point` datatype and matching the result back into Symbolic representation

Reply:
(And as for `Pairing` class, I'd generalize its definition to arbitrary groups as pairing is defined for arbitrary groups anyway. This would also simplify type signatures in `Plonk` quite a bit)

Reply:
Yes, on the second thought, I agree with both points.